tests/test_monte_carlo.c

```diff
@@ -0,0 +1,155 @@
+        }
+      }
+    }
+  }
+}
+
+static int getNumPossibleActions(Board board) {
+  int nActions = 0;
+  for (int x = 0; x < BOARD_LENGTH; ++x) {
+    for (int y = 0; y < BOARD_LENGTH; ++y) {
+      if (board.values[x * BOARD_LENGTH + y] == 0) {
+        nActions++;
+      }
+    }
+  }
+  return nActions * board.nPlayers;
+}
+
+static int getBoardSize() { return 9; }
+
+static bool isDone(Board* board) {
+  for (int i = 0; i < 9; ++i) {
+    if (board->values[i] == 0) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * If there is a line of X or O's
+ * 1 = draw
+ * 2 = win
+ * 3 = win with only 3 plays
+ */
+int getScore(Board* board, int player) {
+  // Check rows
+  for (int i = 0; i < BOARD_SIZE; i += BOARD_LENGTH) {
+    if (board->values[i] == player && board->values[i + 1] == player &&
+        board->values[i + 2] == player) {
+      int nPlays = 0;
+      for (int j = 0; j < 9; ++j) {
+        if (board->values[j] == player) {
+          ++nPlays;
+        }
+      }
+      return (nPlays == 3) ? 3 : 2;
+    }
+  }
+  // Check columns
+  for (int i = 0; i < BOARD_LENGTH; i++) {
+    if (board->values[i] == player && board->values[i + BOARD_LENGTH] == player &&
+        board->values[i + 6] == player) {
+      int nPlays = 0;
+      for (int j = 0; j < 9; ++j) {
+        if (board->values[j] == player) {
+          ++nPlays;
+        }
+      }
+      return (nPlays == 3) ? 3 : 2;
+    }
+  }
+  // Check diagonals
+  if ((board->values[0] == player && board->values[4] == player &&
+       board->values[8] == player) ||
+      (board->values[2] == player && board->values[4] == player &&
+          board->values[6] == player)) {
+    int nPlays = 0;
+    for (int j = 0; j < 9; ++j) {
+      if (board->values[j] == player) {
+        ++nPlays;
+      }
+    }
+    return (nPlays == 3) ? 3 : 2;
+  }
+  return 1;
+}
+
+void removeAction(int randomActionIdx, Action* possibleActions,
+                  int nPossibleActions) {
+  if (randomActionIdx != nPossibleActions + 1) {
+    int secondAction = 0;
+    if (possibleActions[randomActionIdx].player == -1) {
+      secondAction = -1;
+    }
+    for (int i = randomActionIdx + secondAction; i < nPossibleActions; ++i) {
+      possibleActions[i] = possibleActions[i + 2];
+    }
+  }
+}
+
+int testMC(int minSimulation, int maxSimulation, int targetScore) {
+  Board board;
+  board.values = calloc(9, sizeof(int));
+  board.values[0] = 1;
+  board.values[1] = 1;
+  board.nPlayers = 2;
+
+  Game game = {
+      isValidAction,         playAction,   getScore,     getPossibleActions,
+      getNumPossibleActions, removeAction, getBoardSize, isDone, 0, 1
+  };
+
+  Action action =
+      mcGame(board, 1, game, minSimulation, maxSimulation, targetScore);
+  free(board.values);
+  if (action.xPos == 0 && action.yPos == 2) {
+    printf("Success : %s()\n", __func__);
+    return 0;
+  } else {
+    printf("Fail : %s(), expected best action to be: Player: %d, [%d, %d]\n",
```

Comment:
If the test fails, you should also display the action returned by the monte carlo algorithm